src/index.js

```diff
@@ -0,0 +1,172 @@
+
+    return axios.get(url, {params:{q:location}})
+    .then((response)=>{
+
+    let latitude = response.data[0].lat;
+    let longitude = response.data[0].lon
+    console.log(longitude)
+    console.log(latitude)
+    return {latitude:latitude, longitude: longitude};
+
+
+    })
+    .catch( (error) => {
+        console.log('error in findLatitudeAndLongitude!');
+      });
+    
+ }
+
+const getWeather = (lat, lon) =>{
+    return axios
+    .get('http://127.0.0.1:5000/weather', {params:{lat:lat,lon:lon}})
+    .then((response)=>{
+       
+        let result = response.data['main']['temp'];
+        return result
+       
+    })
+    .catch((error)=>{
+        console.log(error);
+    }
+
+    )
+}
+
+const getLocationWeather = () =>{
+    let tempVal = document.getElementById("tempValue");
+    return getLocation()
+    .then( result =>{
+             getWeather(result.latitude, result.longitude)
+            .then(
+                (response)=>{
+                  
+                    let farenheit = Math.floor(((response) - 273.15) * 9/5 + 32 )
+                    console.log(farenheit);
+                    tempVal.textContent = farenheit;
+                    return farenheit
+
+                }
+            )
+          
+          
+            
+        }
+    )
+   
+}
+
+
+
+
+
+// states
+const state = {
+    temp: 0,
+    landscape:"🌲🌲⛄️🌲⛄️🍂🌲🍁🌲🌲⛄️🍂🌲",
+    city: "Seattle"
+
+}
+
+// selected elements
+const temp = document.getElementById("tempValue");
+const controlUp = document.getElementById("increaseTempControl");
+const controlDown = document.getElementById("decreaseTempControl");
+const landscape = document.getElementById("landscape");
+const cityName = document.getElementById("headerCityName");
+const cityInput = document.getElementById("cityNameInput");
+const reset = document.getElementById("cityNameReset");
+const skyImage = document.getElementById("sky");
+const sky = document.getElementById("skySelect");
+const weatherGarden = document.getElementById("gardenContent")
+const tempbutton = document.getElementById("currentTempButton")
+
+// helper functions
+const changeColor = () =>{
+    if (state.temp>80){
+        temp.className = 'red';
+        landscape.innerText = "🌵__🐍_🦂_🌵🌵__🐍_🏜_🦂";
+    }
+    else if (state.temp>69){
+        temp.className = 'orange';
+        landscape.innerText = "🌸🌿🌼__🌷🌻🌿_☘️🌱_🌻🌷"
+    }
+    else if(state.temp>59){
+        temp.className = 'yellow';
+        landscape.innerText = "🌾🌾_🍃_🪨__🛤_🌾🌾🌾_🍃"
+    }
+    else if(state.temp>49){
+        temp.className = 'green'
+        landscape.innerText = "🌲🌲⛄️🌲⛄️🍂🌲🍁🌲🌲⛄️🍂🌲"
+    }
+    else{
+        temp.className = 'teal'
+        landscape.innerText = "🌲🌲⛄️🌲⛄️🍂🌲🍁🌲🌲⛄️🍂🌲"
+    }
+}
+
+// functions
+const increaseTemp = () =>{
+    changeColor()
+    state.temp+= 1
+    temp.innerText = state.temp
+    
+}
+
+const decreaseTemp = () => {
+    changeColor()
+    state.temp -= 1
+    temp.innerText = state.temp
```

Comment:
Couple of thoughts here for these two methods.
1) Missing semi colons
2) Do you mean for the changeColor() method to be invoked after you update state.temp and reset temp.innerText? Like this:

```javascript
    state.temp -= 1
    temp.innerText = state.temp
    changeColor()
```

3) Also, notice that these two methods do essentially the same thing but the difference is -= 1 or += 1 (also in JS you can use ++ or -- which is short hand for doing the same thing).

How could you combine the logic of these two methods into one event handler called handleTempChange()?

What if you pass in the event object as an argument and you check if the event's target has an id `increaseTempControl` or `decreaseTempControl` and depending on the id then you could ++ or --

```javascript
const handleTempChange = (event) => {
    if (event.target.id === 'increaseTempControl') {
        state.temp++;
    } else {
        state.temp--;
    }
    temp.innerText = state.temp;
    changeColor();
};